templates/experiments/components/experiment_chat.html

```diff
@@ -1,103 +1,158 @@
 {% load json_tags %}
 {% load chat_tags %}
 {% load waffle_tags %}
+{% load pagination_tags %}
 
-<div x-data="{search: null}">
-  {% if perms.annotations.view_customtaggeditem %}
-    <div class="mt-1 max-w-5xl">
-      <input type="text" x-model="search" class="input input-bordered input-sm w-1/2" placeholder="Search tags...">
+<div id="table-top" class="max-w-5xl mt-4 border rounded-2xl">
+  <div class="mt-1 max-w-5xl flex justify-between p-3">
+    <div class="flex flex-col gap-2">
+      {% if perms.annotations.view_customtaggeditem %}
+        <div class="flex items-center gap-2">
+          <input
+            type="text"
+            name="search"
+            value="{{ search|default:'' }}"
+            class="input input-bordered input-sm w-96"
+            placeholder="Search tags..."
+            hx-get="{% url 'experiments:experiment_session_messages_view' request.team.slug experiment.public_id experiment_session.external_id %}"
+            hx-trigger="keyup changed delay:500ms"
+            hx-target="#messages-container"
+            hx-include="[name='search']"
+          >
+          {% if search %}
+            <button
+              class="btn btn-ghost btn-sm"
+              hx-get="{% url 'experiments:experiment_session_messages_view' request.team.slug experiment.public_id experiment_session.external_id %}"
+              hx-target="#messages-container"
+            >
+              <i class="fa-solid fa-times"></i> Clear
```

Comment:
nit: remove 'Clear' to be consistent with other UIs
```suggestion
              <i class="fa-solid fa-times"></i>
```